protocol/protocol.go

```diff
@@ -625,6 +625,8 @@ func CreateReplies(ver Version, requests []Request, midpoint time.Time, radius t
 		binary.LittleEndian.PutUint32(indexBytes[:], uint32(i))
 		reply[tagINDX] = indexBytes[:]
 
+		reply[tagNONC] = requests[i].Nonce
```

Comment:
Thanks! Looks like the Google-Roughtime it's not required to send `NONC` back, but also not forbidden: https://roughtime.googlesource.com/roughtime/+/refs/heads/master/PROTOCOL.md#responses, so this seems OK.

We claim at https://github.com/cloudflare/roughtime/blob/master/protocol/protocol.go#L811 that `VerifyReply` checks that the server response includes a nonce, but it doesn't look like we actually perform that check (which is why the current client/server implementations in this repo interoperate).

@dansarie would you want to add that check in to this PR, perhaps at https://github.com/cloudflare/roughtime/blob/master/protocol/protocol.go#L969?

Reply:
@lukevalenta Okay! Should be fixed now.